src/mpi/pt2pt/mpir_request.c

```diff
@@ -547,38 +547,12 @@ int MPIR_Grequest_progress_poke(int count,
 {
     MPIX_Grequest_wait_function *wait_fn = NULL;
     void ** state_ptrs;
-    int i, j, n_classes, n_native, n_greq;
+    int i, j, n_greq;
     int mpi_errno = MPI_SUCCESS;
     MPIR_CHKLMEM_DECL(1);
 
     MPIR_CHKLMEM_MALLOC(state_ptrs, void **, sizeof(void*) * count, mpi_errno, "state_ptrs");
 
-    /* This somewhat messy for-loop computes how many requests are native
-     * requests and how many are generalized requests, and how many generalized
-     * request classes those grequests are members of */
-    for (i=0, j=0, n_classes=1, n_native=0, n_greq=0; i< count; i++)
-    {
-	if (request_ptrs[i] == NULL || MPIR_Request_is_complete(request_ptrs[i])) continue;
-	if (request_ptrs[i]->kind == MPIR_REQUEST_KIND__GREQUEST)
-	{
-	    n_greq += 1;
-	    wait_fn = request_ptrs[i]->u.ureq.greq_fns->wait_fn;
-	    state_ptrs[j] = request_ptrs[i]->u.ureq.greq_fns->grequest_extra_state;
-	    j++;
-	    if (i+1 < count) {
-	        if (request_ptrs[i+1] == NULL ||
-			(request_ptrs[i]->u.ureq.greq_fns->greq_class !=
-				request_ptrs[i+1]->u.ureq.greq_fns->greq_class) )
-	            n_classes += 1;
-	    }
-	} else {
-	    n_native += 1;
-	}
-    }
-
-    if (j > 0 && n_classes == 1 && wait_fn != NULL) {
-        mpi_errno = (wait_fn)(j, state_ptrs, 0, NULL);
-    } else {
 	for (i = 0; i< count; i++ )
```

Comment:
We should make sure to re-align this lines.

Reply:
it's a tab vs. space problem. The files I touched were full of these type of inconsistencies. I can pass the code cleanup script on those files if you want

Reply:
I very  much like the simpler request handling.  The tradeoff is the removal of one optimization.  

The idea of a "class" of generalized requests exists so that we can complete multiple requests at once.  `aio_suspend` can complete up `nitems` finishing within `timeout`.     So I looped over all outstanding requests, collecting them into their classes, then completed that class.  

Switching to testing each request is correct , and correct always beats fast.    There was (a decade ago) a benefit to doing class-wise completion, but this is not common code and should not muck up the more common code paths.

Reply:
@roblatham00 , I added a TODO and updated the commit message to reference the class-wise completion